subprojects/docs/src/samples/userguide/dependencyManagement/workingWithDependencies/walkGraph/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,46 @@
+repositories {
+    jcenter()
+}
+
+val scm by configurations.creating
+
+dependencies {
+    scm("org.eclipse.jgit:org.eclipse.jgit:4.9.2.201712150930-r")
+    scm("commons-codec:commons-codec:1.7")
+    scm("some:unresolved:2.5")
+}
+
+// tag::walk-task[]
+task("walkDependencyGraph", DependencyGraphWalk::class) {
+    dependsOn(configurations["scm"])
+}
+
+open class DependencyGraphWalk: DefaultTask() {
+    @TaskAction
+    fun walk() {
+        val configuration: Configuration = project.configurations.getByName("scm")
+        val resolutionResult: ResolutionResult = configuration.incoming.resolutionResult
+        val root: ResolvedComponentResult = resolutionResult.root
+        logger.quiet(configuration.name)
+        traverseDependencies(0, root.dependencies)
+    }
+
+    private fun traverseDependencies(level: Int, results: Set<DependencyResult>) {
+        results.forEach { result ->
+            if (result is ResolvedDependencyResult) {
```

Comment:
⭕️ This `if/else if` would read better as `when`

Reply:
I'll pass here: @eriwen just told me to stay close to the groovy sample and avoid kotlinisms.